src/utils/functions/signs.ts

```diff
@@ -0,0 +1,17 @@
+import { Sign } from '../../state-manager/state-manager-types'
+
+export function removeDuplicateSignatures(signatures: Sign[]): Sign[] {
+  const seen = new Set<string>()
+  const uniqueSignatures = []
+
+  for (const sign of signatures) {
+    const key = `${sign.owner}-${sign.sig}`
```

Comment:
Checking the presence of the owner in the set should be good enough. As one signer can only produce one valid payload for a given payload.

Reply:
Done